aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+
+.. code-block:: html
+
+  <p>@DateTime.Now</p>
+  <p>@DateTime.IsLeapYear(2016)</p>
+
+Implicit expressions generally cannot contain spaces. For example, in the code below, one week is not subtracted from the current time:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @* End of greeting *@
+  :end-before: @*Add () to get correct time.*@
+
+Which renders the following HTML:
+
+.. code-block:: html
+
+  <p>Last week: 7/7/2016 4:39:52 PM - TimeSpan.FromDays(7)</p>
+
+Using parenthesis fixes the problem:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @*Add () to get correct time.*@
+  :end-before: @*End of correct time*@
+
+.. tip:: You can use C# expressions in Razor that contain spaces using an :ref:`explicit expression <explicit-razor-expressions>`.
+  
+With the exception of the C# ``await`` keyword implicit expressions must not contain spaces. For example, you can intermingle spaces as long as the C# statement has a clear ending:
+
+.. code-block:: html
+
+  <p>@await DoSomething("hello", "world")</p>
+
+.. _explicit-razor-expressions:
+
+Explicit Razor expressions
+----------------------------
+
+Explicit Razor expressions consists of an @ symbol with balanced parenthesis. For example, to render last weeks’ time:
+
+.. code-block:: html
+
+  <p>Last week this time: @(DateTime.Now - TimeSpan.FromDays(7))</p>
+
+Any content within the @() parenthesis is evaluated and rendered to the output.
+
+.. _expression-encoding-label:
+
+Expression encoding
+-------------------
+
+Non-:dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent` content is HTML encoded. For example, the following Razor markup:
+
+.. code-block:: html
+
+  @("<span>Hello World</span>")
+
+Renders this HTML:
+
+.. code-block:: html
+
+  &lt;span&gt;Hello World&lt;/span&gt;
+
+:dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` :dn:method:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.Raw` output is not encoded but rendered as HTML markup. :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` implements :dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent`
+
+.. warning:: Using ``HtmlHelper.Raw`` on unsanitzed user input is a security risk. User input might contain malicious JavaScript or other exploits. Sanitizing user input is difficult, avoid using ``HtmlHelper.Raw`` on user input.
+
+The following Razor markup:
+
+.. code-block:: html
+
+  @Html.Raw("<span>Hello World</span>")
+
+Renders this HTML:
+
+.. code-block:: html
+
+  <span>Hello World</span>
+
+Which the browser renders as:
+``Hello World``
+
+.. _razor-code-blocks-label:
+
+Razor code blocks
+------------------
+
+Razor code blocks start with ``@`` and are enclosed by ``{}``. Unlike expressions, C# code inside code blocks is not rendered.
+
+.. comment: sample shows var defined in first code block available in 2nd block. Some customer confusion on that.
```

Comment:
This sample seems overly complicated. It shows implicit transitions, but those haven't been introduced yet. I prefer the simpler sample @NTaylorMullen had originally:

```
@{
    var output = "Hello World";
}

<p>The rendered result: @output</p>
```

Would render:

```html
<p>The rendered result: Hello World</p>
```